sa/storage-authority.go

```diff
@@ -271,3 +343,34 @@ func (ssa *SQLStorageAuthority) FinalizeAuthorization(authz core.Authorization)
 	err = tx.Commit()
 	return
 }
+
+func (ssa *SQLStorageAuthority) AddCertificate(cert []byte) (id string, err error) {
+	tx, err := ssa.db.Begin()
+	if err != nil {
+		return
+	}
+
+	// Manually set the index, to avoid AUTOINCREMENT issues
+	var sequence int64
+	var scanTarget sql.NullInt64
+	err = tx.QueryRow("SELECT max(sequence) FROM certificates;").Scan(&scanTarget)
+	switch {
+	case !scanTarget.Valid:
+		sequence = 0
+	case err != nil:
+		tx.Rollback()
+		return
+	default:
+		sequence += scanTarget.Int64 + 1
+	}
+
+	id = core.Fingerprint256(cert)
+	_, err = tx.Exec("INSERT INTO certificates (sequence, digest, value) VALUES (?,?,?);", sequence, id, cert)
```

Comment:
After further thought I think there may even be a way to do this lockless, where the INSERT query includes the SELECT max() query, but it's just a nit. I look forward to the day when we start to care about one lock!
(Something like: "INSERT INTO certificates (...) VALUES ((SELECT max(sequence) FROM certificates), ?, ?);" )